### PR TITLE
SC: Added handledRequestValueTransfer<hash, isDone> map into bridge contract.

### PR DESCRIPTION
**Pull request by KimKyungup**
## Proposed changes
This PR added handledRequestValueTransfer<Hash, isDone> map into bridge contract.
Through this PR user can know which request is processed or not.

## Types of changes

Please put an x in the boxes related to your change.

- [ ] Bugfix
- [x] New feature or enhancement
- [ ] Others

## Checklist

*Put an x in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code.*

- [x] I have read the [CONTRIBUTING GUIDELINES](https://github.com/klaytn/klaytn/blob/master/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://cla-assistant.io/klaytn/klaytn)
- [x] Lint and unit tests pass locally with my changes (`$ make test`)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: contracts/bridge/HandledRequests.sol [hunk 1]
```diff
@@ -0,0 +1,10 @@
+pragma solidity ^0.4.24;
+
+contract HandledRequests {
+    // TODO-Klaytn-Servicechain handleTxHash can be saved after Klaytn supports it.
+    mapping (bytes32 => bool) public isHandledRequestTx;
```
Comment (jeonghonah) [inline contracts/bridge/HandledRequests.sol]: isHandledRequestTx --> handledRequestTxs

File: contracts/bridge/Bridge.go [hunk 2]
```diff
@@ -525,6 +525,32 @@ func (_Bridge *BridgeCallerSession) HandleNonce() (uint64, error) {
 	return _Bridge.Contract.HandleNonce(&_Bridge.CallOpts)
 }
 
+// HandledRequestTx is a free data retrieval call binding the contract method 0x8a75eee2.
+//
+// Solidity: function handledRequestTx( bytes32) constant returns(bool)
```
Comment (aidan-kwon) [inline contracts/bridge/Bridge.go]:
```suggestion
// Solidity: function handledRequestTx(bytes32) constant returns(bool)
```
Please, correct the same typo in other lines also.

Comment (KimKyungup) [reply contracts/bridge/Bridge.go]: The go file is automatically generated by abigen. But I will try to find how to fix it.

Comment (KimKyungup) [reply contracts/bridge/Bridge.go]: @aidan-kwon This template code is in `github.com/klaytn/klaytn/accounts/abi/bind/template.go`.
I can't find the solution now. I will consider it later. :)

File: contracts/bridge/HandledRequests.sol [hunk 3]
```diff
@@ -0,0 +1,10 @@
+pragma solidity ^0.4.24;
+
+contract HandledRequests {
+    // TODO-Klaytn-Servicechain handleTxHash can be saved after Klaytn supports it.
+    mapping (bytes32 => bool) public handledRequestTx;
```
Comment (aidan-kwon) [inline contracts/bridge/HandledRequests.sol]:
```suggestion
    mapping(bytes32 => bool) public handledRequestTx;
```

File: contracts/bridge/Bridge.go (hunk 2 above)
Comment (aidan-kwon) [reply contracts/bridge/Bridge.go]: @KimKyungup Thank you. I resolve the conversation.

**Comment by KimKyungup** [comment]
@aidan-kwon PTAL. :)

File: contracts/bridge/Bridge.sol [hunk 4]
```diff
@@ -122,7 +126,8 @@ contract Bridge is IERC20BridgeReceiver, IERC721BridgeReceiver, Ownable, BridgeF
     {
         require(handleNonce == _requestNonce, "mismatched handle / request nonce");
 
-        emit HandleValueTransfer(TokenType.ERC20, _from, _to, _tokenAddress, _value, handleNonce);
+        emit HandleValueTransfer(_requestTxHash, TokenType.ERC20, _from, _to, _tokenAddress, _value, handleNonce);
+        _setHandledRequestTxHash(_requestTxHash);
```
Comment (aidan-kwon) [inline contracts/bridge/Bridge.sol]: How about switching the order of L129 and L130?
Just to ensure an event has occurred after a value set. 
(And, the same comment for L156, L180)

Comment (KimKyungup) [reply contracts/bridge/Bridge.sol]: Solidity event will be canceled if it is reverted. :)

**Comment by KimKyungup** [comment]
@aidan-kwon @jeonghonah Rebased. :)

**Pull request merged**